File: crime-analytics-frontend/src/services/crimeService.js
import axiosInstance from '../api/axiosInstance';

export const crimeService = {
  async getAllCrimes() {
    const response = await axiosInstance.get('/crimes');
    return response.data;
  },

  async getCrimeById(id) {
    const response = await axiosInstance.get(`/crimes/${id}`);
    return response.data;
  },

  async createCrime(data) {
    const response = await axiosInstance.post('/crimes', data);
    return response.data;
  },

  async updateCrime(id, data) {
    const response = await axiosInstance.put(`/crimes/${id}`, data);
    return response.data;
  },

  async deleteCrime(id) {
    const response = await axiosInstance.delete(`/crimes/${id}`);
    return response.data;
  }
};

File: crime-analytics-backend/src/main/java/com/crimeanalytics/service/CrimeService.java
package com.crimeanalytics.service;

import com.crimeanalytics.dto.CrimeRequestDTO;
import com.crimeanalytics.dto.CrimeResponseDTO;

import java.util.List;

public interface CrimeService {

    CrimeResponseDTO createCrime(CrimeRequestDTO requestDTO);

    CrimeResponseDTO getCrimeById(Long id);

    List<CrimeResponseDTO> getAllCrimes();

    CrimeResponseDTO updateCrime(Long id, CrimeRequestDTO requestDTO);

    void deleteCrime(Long id);
}